mino/minogrpc/server.go

```diff
@@ -122,19 +122,35 @@ func (o overlayServer) Join(ctx context.Context, req *ptypes.JoinRequest) (*ptyp
 }
 
 // Share implements ptypes.OverlayServer. It accepts a certificate from a
-// participant.
+// participant only if it is valid from the address it claims to be.
 func (o overlayServer) Share(ctx context.Context, msg *ptypes.Certificate) (*ptypes.CertificateAck, error) {
-	// TODO: verify the validity of the certificate by connecting to the distant
-	// node but that requires a protection against malicious share.
-
-	from := o.addrFactory.FromText(msg.GetAddress())
+	from := o.addrFactory.FromText(msg.GetAddress()).(address)
 
 	cert, err := x509.ParseCertificate(msg.GetValue())
 	if err != nil {
 		return nil, xerrors.Errorf("couldn't parse certificate: %v", err)
 	}
 
-	o.certs.Store(from, &tls.Certificate{Leaf: cert})
+	// Make sure the certificate is valid for the public key provided.
+	err = cert.CheckSignature(cert.SignatureAlgorithm, cert.RawTBSCertificate, cert.Signature)
```

Comment:
apparently its the same as doing `cert.CheckSignatureFrom(cert)`, but it adds few more checks.

Reply:
It does not accept the same certificate as a valid parent.

Reply:
nevermind, it's a question of configuration.